index.bs

```diff
@@ -402,9 +565,29 @@ perform the following steps:
 
 </div>
 
-Every [=bounce tracking timer period=] the user agent should run the
-[=bounce tracking timer=] algorithm given the [=wall clock=]'s
-[=wall clock/unsafe current time=].
+This algorithm is called from [=process response received for bounce tracking=],
+after a timeout of [=client bounce detection timer period=], unless it is cancelled
+by toggling [=bounce tracking record/client bounce detection running=].
+
+<div algorithm>
+
+To run the <dfn>record stateful bounces for bounce tracking</dfn> algorithm
+given a [=navigable=] |navigable|, perform the following steps:
+
+1. If |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/client bounce detection running=] is false, abort these steps.
+1. [=map/For each=] |host| in |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/bounce set=]:
+    1. If |host| [=host/equals=] |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/initial host=], [=iteration/continue=].
+    1. If |host| [=host/equals=] |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/final host=], [=iteration/continue=].
+    1. If [=user activation map=] [=map/contains=] |host|, [=iteration/continue=].
+    1. If |navigable|'s [=top-level traversable/bounce tracking record=]'s
+        [=bounce tracking record/storage access set=] does not [=set/contain=] |host|, [=iteration/continue=].
+    1. Let |topDocument| be |navigable|'s [=top-level traversable=]'s
+        [=navigable/active document=].
+    1. Set [=stateful bounce tracking map=][|host|] to |topDocument|'s
+        [=relevant settings object=]'s [=environment settings object/current wall time=].
```

Comment:
I don't think this implements the "first wall clock time since the last execution of the bounce tracking timer" bit of the map's definition.

Reply:
The full definition is "first wall clock time since the last execution of the bounce tracking timer at which... stateful bounce tracking took place"

When bounces are detected and recorded here, we set the map value to the current wall time. Later when the bounce tracking timer algorithm runs, it clears the map.

Reply:
Oh, I see, thanks for the catch.

@wanderview is this definition correct? In our implementation we always update the bounce timestamp when we detect a new stateful bounce, so the map ends up storing the last wall clock time that stateful bounce tracking took place. Not sure if the distinction matters though.

Reply:
Oh right, thanks for the catch.

@wanderview I don't think this definition matches our implementation, we update the bounce timestamp after every new stateful bounce, so our map actually stores the *last* wall clock time at which stateful bounce tracking took place.

Is this distinction important? We could easily switch to *first* by aborting this algorithm if the map already has a timestamp, and that would remove duplicate handling of the same site.